{
  "affected": [
    {
      "database_specific": {
        "source": "https://storage.googleapis.com/cve-osv-conversion/osv-output/CVE-2021-29617.json"
      },
      "ranges": [
        {
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "890f7164b70354c57d40eda52dcdd7658677c09f"
            }
          ],
          "repo": "https://github.com/tensorflow/tensorflow",
          "type": "GIT"
        }
      ],
      "versions": [
        "0.12.0-rc0",
        "0.12.0-rc1",
        "0.12.1",
        "0.5.0",
        "0.6.0",
        "v0.10.0",
        "v0.10.0rc0",
        "v0.11.0",
        "v0.11.0rc0",
        "v0.11.0rc1",
        "v0.11.0rc2",
        "v0.12.0",
        "v0.7.0",
        "v0.7.1",
        "v0.8.0rc0",
        "v0.9.0",
        "v0.9.0rc0",
        "v1.0.0",
        "v1.0.0-alpha",
        "v1.0.0-rc0",
        "v1.0.0-rc1",
        "v1.0.0-rc2",
        "v1.1.0",
        "v1.1.0-rc0",
        "v1.1.0-rc1",
        "v1.1.0-rc2",
        "v1.12.0",
        "v1.12.0-rc0",
        "v1.12.0-rc1",
        "v1.12.0-rc2",
        "v1.12.1",
        "v1.2.0",
        "v1.2.0-rc0",
        "v1.2.0-rc1",
        "v1.2.0-rc2",
        "v1.3.0-rc0",
        "v1.3.0-rc1",
        "v1.5.0",
        "v1.5.0-rc0",
        "v1.5.0-rc1",
        "v1.6.0",
        "v1.6.0-rc0",
        "v1.6.0-rc1",
        "v1.7.0",
        "v1.7.0-rc0",
        "v1.7.0-rc1",
        "v1.8.0",
        "v1.8.0-rc0",
        "v1.8.0-rc1",
        "v1.9.0",
        "v1.9.0-rc0",
        "v1.9.0-rc1",
        "v1.9.0-rc2"
      ]
    }
  ],
  "aliases": [
    "BIT-tensorflow-2021-29617",
    "GHSA-mmq6-q8r3-48fm",
    "PYSEC-2021-254",
    "PYSEC-2021-545",
    "PYSEC-2021-743"
  ],
  "details": "TensorFlow is an end-to-end open source platform for machine learning. An attacker can cause a denial of service via `CHECK`-fail in `tf.strings.substr` with invalid arguments. The fix will be included in TensorFlow 2.5.0. We will also cherrypick this commit on TensorFlow 2.4.2, TensorFlow 2.3.3, TensorFlow 2.2.3 and TensorFlow 2.1.4, as these are also affected and still in supported range.",
  "id": "CVE-2021-29617",
  "modified": "2025-10-14T18:27:51.124339Z",
  "published": "2021-05-14T20:15:16Z",
  "references": [
    {
      "type": "FIX",
      "url": "https://github.com/tensorflow/tensorflow/commit/890f7164b70354c57d40eda52dcdd7658677c09f"
    },
    {
      "type": "FIX",
      "url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-mmq6-q8r3-48fm"
    },
    {
      "type": "REPORT",
      "url": "https://github.com/tensorflow/issues/46900"
    },
    {
      "type": "REPORT",
      "url": "https://github.com/tensorflow/issues/46974"
    }
  ],
  "related": [
    "GHSA-mmq6-q8r3-48fm"
  ],
  "schema_version": "1.7.3",
  "severity": [
    {
      "score": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H",
      "type": "CVSS_V3"
    }
  ]
}